message.go

```diff
@@ -523,24 +531,51 @@ func (r *prettyRedisResult) MarshalJSON() ([]byte, error) {
 
 // RedisMessage is a redis response message, it may be a nil response
 type RedisMessage struct {
-	attrs   *RedisMessage
-	string  string
-	values  []RedisMessage
+	attrs *RedisMessage
+	bytes *byte
+	array *RedisMessage
+
+	// integer is used for a simple number or
+	// in conjunction with array or bytes to store the length of array or string
 	integer int64
 	typ     byte
 	ttl     [7]byte
 }
 
+func (m *RedisMessage) string() string {
+	if m.bytes == nil {
+		return ""
+	}
+	return unsafe.String(m.bytes, m.integer)
+}
+
+func (m *RedisMessage) values() []RedisMessage {
```

Comment:
```suggestion
func (m RedisMessage) values() []RedisMessage {
```
Is there any performance downside to using a value receiver on this function? Can this function be inlined?